### DISCUSSION TITLE
How to export OpenAPI Schema for use in docs or as a static site

### DISCUSSION THREAD

**Discussion by codekiln** [cookiecutter/cookiecutter-django#4417] [Q&A]
I see that when DRF is enabled, this appears in base.py: 

```
# By Default swagger ui is available only to admin user(s). You can change permission classes to change that
# See more configuration options at https://drf-spectacular.readthedocs.io/en/latest/settings.html#settings
SPECTACULAR_SETTINGS = {
    "TITLE": "<proj title> API",
    "DESCRIPTION": "Documentation of API endpoints of <proj title>",
    "VERSION": "1.0.0",
    "SERVE_PERMISSIONS": ["rest_framework.permissions.IsAdminUser"],
}
```

I'm able to access the page at http://localhost:3000/api/docs/# when logged in as an admin, even though I didn't see this referenced in the docs anywhere (I found it via https://github.com/cookiecutter/cookiecutter-django/issues/3593#issue-1139170435)
![image](https://github.com/cookiecutter/cookiecutter-django/assets/140930/5d7de818-9c8d-4c86-9637-5b028cac5f6d)

Does anyone know if there's an easy way to export the API docs so they appear as part of the Sphynx docs, or if one can export the OpenAPI schema for generating a documentation site outside of Sphynx? I already tried `docker-compose -f local.yml run --rm docs make apidocs` and then `docker-compose -f local.yml up docs`, but I didn't see any included API documentation.

I do see the [Take it for a spin section](https://drf-spectacular.readthedocs.io/en/latest/readme.html#take-it-for-a-spin) of drf-spectacular says one can generate your schema with `./manage.py spectacular --color --file schema.yml` 

... 
Actually, I just figured this out: 

`docker-compose -f local.yml run --rm django python manage.py spectacular --color --file schema.yml` will generate a `schema.yml` that's an OpenAPI schema. Hope this helps someone else!

**Comment by browniebroke** [comment]
You might want to post the solution as a separate message and mark it as answer. It will flag the question as "answered" in the list, making it clearer to others that there is a solution 😉
